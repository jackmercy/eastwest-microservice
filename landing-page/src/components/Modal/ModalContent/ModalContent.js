import React, { memo } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import useStyles from './modalContent.styles';

const ModalContent = ({ children, className, ...restProps }) => {
  const classes = useStyles();

  return (
    <div
      className={classNames('custom-scrollbar', classes.root, className)}
      {...restProps}
    >
      {children}
    </div>
  );
};
ModalContent.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};
ModalContent.defaultProps = { className: '' };

export default memo(ModalContent);
