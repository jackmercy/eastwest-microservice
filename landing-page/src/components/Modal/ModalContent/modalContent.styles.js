import { createUseStyles } from 'react-jss';

import { rem } from '../../../utils/jss';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

export default createUseStyles({
  root: {
    flex: '1 1 auto',
    overflow: 'auto',
    padding: rem(32),
    transition: 'padding .5s ease',
    fontSize: rem(24),
    lineHeight: rem(32),
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    root: { padding: rem(16), fontSize: rem(14), lineHeight: rem(20) },
  },
});
