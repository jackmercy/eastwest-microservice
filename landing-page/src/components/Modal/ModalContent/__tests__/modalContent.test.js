import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ModalContent from '../ModalContent';

describe('Testing Modal component > ModalContent', () => {
  it('should render match snapshot', () => {
    const wrapper = shallow(<ModalContent>Content</ModalContent>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
