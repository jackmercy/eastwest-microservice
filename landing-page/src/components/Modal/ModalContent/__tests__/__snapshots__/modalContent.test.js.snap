// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Modal component > ModalContent should render match snapshot 1`] = `
<div
  className="custom-scrollbar root-0-2-1"
>
  Content
</div>
`;
